Converted Models from Java to Kotlin on convertModelsBranch

adityapatel-sudo opened the pull request:
Changed Flashcard and Deck Models into Kotlin from Java.

adityapatel-sudo committed:
Converted Models from Java to Kotlin on convertModelsBranch

adityapatel-sudo committed:
Converted Models from Java to Kotlin on convertModelsBranch
Addressed:
-can we align it in multiple lines for readability
-can we have a null deckName , I feel we should not make it nullable
-I think we need both work and definition , making it nullable wouldnt help.
-add a newline at the end of file

swatisaoji1 commented on app/src/main/java/com/adityap/flashy_createflashcards/FlashcardDatabaseHelper.kt:
```diff
@@ -67,10 +69,7 @@ class FlashcardDatabaseHelper  //private static final String FOREIGN_KEY_LINK =
-                flashcard.setmId(cursor.getInt(cursor.getColumnIndex(DECK_ID)))
-                flashcard.word = cursor.getString(cursor.getColumnIndex(WORD))
-                flashcard.definition = cursor.getString(cursor.getColumnIndex(DEFINITION))
+                val flashcard = FlashcardModel(cursor.getInt(cursor.getColumnIndex(DECK_ID)), cursor.getString(cursor.getColumnIndex(WORD)), cursor.getString(cursor.getColumnIndex(DEFINITION)))
```
can we align it in multiple lines for readability

swatisaoji1 commented on app/src/main/java/com/adityap/flashy_createflashcards/models/DeckModel.kt:
```diff
@@ -0,0 +1,34 @@
+/**
+ * Created by Aditya on 12/26/2018.
+ */
+data class DeckModel(val id : Int = 0, val deckName: String?, val deckDescription: String?) : Parcelable {
```
can we have a null deckName , I feel we should not make it nullable

swatisaoji1 commented on app/src/main/java/com/adityap/flashy_createflashcards/models/FlashcardModel.kt:
```diff
@@ -0,0 +1,35 @@
+/**
+ * Created by Aditya on 9/30/2018.
+ */
+data class FlashcardModel(val id: Int = 0, val word: String?, val definition: String?) : Parcelable {
```
I think we need both work and definition , making it nullable wouldnt help.

swatisaoji1 commented on app/src/main/java/com/adityap/flashy_createflashcards/models/FlashcardModel.kt:
```diff
@@ -0,0 +1,35 @@
+            return arrayOfNulls(size)
+        }
+    }
+}
```
add a newline at the end of file

swatisaoji1 reviewed (changes requested):
Some nits mostly

adityapatel-sudo committed:
Merge remote-tracking branch 'github/convertModelsBranch' into convertModelsBranch

# Conflicts:
#	app/src/main/java/com/adityap/flashy_createflashcards/FlashcardDatabaseHelper.kt
#	app/src/main/java/com/adityap/flashy_createflashcards/models/DeckModel.kt
#	app/src/main/java/com/adityap/flashy_createflashcards/models/FlashcardModel.kt

adityapatel-sudo commented:
* Converted Models from Java to Kotlin on convertModelsBranch
Addressed:
-can we align it in multiple lines for readability
-can we have a null deckName , I feel we should not make it nullable
-I think we need both work and definition , making it nullable wouldnt help.
-add a newline at the end of file
